Cisco/cisco-esa/ingest/parser.yml

```
@@ -209,9 +209,14 @@ stages:
                 {% endif %}
               {% endif %}
             {%- endfor %}]
-          cisco.esa.url: >-
-            [{% for url, details in dict(json_event_url_details.message).items() %}
-              "{% if details.get('ExpandedUrl') is not none %}{{ details.ExpandedUrl }}{% else %}{{ url }}{% endif %}"
+          cisco.esa.url.domain: >-
+            [{% for url, details in json_event_url_details.message.items() %}
+              {% if details.get('ExpandedUrl') is not none %}"{{url.split('/')[2]}}", "{{ details.ExpandedUrl.split('/')[2] }}"{% else %}"{{ url.split('/')[2] }}"{% endif %}
+              {% if not loop.last %},{% endif %}
+            {% endfor %}]
+          cisco.esa.url.full: >-
+            [{% for url, details in json_event_url_details.message.items() %}
+              {% if details.get('ExpandedUrl') is not none %}"{{url}}", "{{ details.ExpandedUrl }}"{% else %}"{{ url }}"{% endif %}
```

Changing the type of field (i.e, `array` -> `dictionary`) will generate some issue in the indexing.
Keep `cisco.esa.url` and rename `cisco.esa.url.domain` into `cisco.esa.url_domain`:
```suggestion
          cisco.esa.url_domain: >-
            [{% for url, details in json_event_url_details.message.items() %}
              {% if details.get('ExpandedUrl') is not none %}"{{url.split('/')[2]}}", "{{ details.ExpandedUrl.split('/')[2] }}"{% else %}"{{ url.split('/')[2] }}"{% endif %}
              {% if not loop.last %},{% endif %}
            {% endfor %}]
          cisco.esa.url: >-
            [{% for url, details in json_event_url_details.message.items() %}
              {% if details.get('ExpandedUrl') is not none %}"{{url}}", "{{ details.ExpandedUrl }}"{% else %}"{{ url }}"{% endif %}
```